### PR TITLE
Home work 5

### PR DESCRIPTION
**Pull request by VolodiaH**

### CODE REVIEW

File: Hw5/main.cpp [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	task1();
+	task2();
+	task3();
+	task4();
+	task5();
+}
+
+void task1() 
+{
+	std::cout << "Enter init value, step, end value:";
```
Comment (helen-black) [inline Hw5/main.cpp]: In this task you have not the last value, but number of the last element. So you can have input: init - 2, step - 100, number of the last - 5.
Fix your code, please. It doesn't work well

File: Hw5/main.cpp [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+long long factorial(int number)
+{
+	if (number < 1)
+		return number;
+
+	long long res{ 1 };
+
+	for (std::size_t i = number; i > 0; --i)
+		res *= i;
```
Comment (helen-black) [inline Hw5/main.cpp:56]: Please, don't miss the brackets in such cases

File: Hw5/main.cpp [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+	std::cout << "\n";
+
+	for (std::size_t i = 0; i < length; ++i)
+		std::cout << std::string(i, ' ') + std::string(length, '*') << '\n';
+
+	std::cout << "\n";
+
+	for (std::size_t i = 1; i <= length; ++i)
+		std::cout << binaryString(i) << '\n';
```
Comment (helen-black) [inline Hw5/main.cpp]: Each row will start with 0. It should start from 0 then from 1...

**Review by helen-black — CHANGES_REQUESTED** [review]

File: Hw5/main.cpp (hunk 1 above)
Comment (VolodiaH) [reply Hw5/main.cpp]: Resolved

File: Hw5/main.cpp (hunk 2 above)
Comment (VolodiaH) [reply Hw5/main.cpp:56]: I fixed narrowing from int to size_t

File: Hw5/main.cpp (hunk 3 above)
Comment (VolodiaH) [reply Hw5/main.cpp]: I guess you were confused by the name of the function, changed it to a more appropriate one

File: Hw5/main.cpp [hunk 4]
```diff
@@ -71,11 +68,11 @@ void task3()
...
 }
 
-std::string binaryString(std::size_t count)
+std::string stripedString(std::size_t count)
 {
 	std::string res("");
 
-	for (std::size_t i = 0; i < count; ++i)
+	for (std::size_t i{ 0 }; i < count; ++i)
 		res.insert(res.begin(), (i & 1) ? '0' : '1');
```
Comment (helen-black) [inline Hw5/main.cpp]: You insert value at the beginning of the string. It's not an effective operation. Try to create better solution

File: Hw5/main.cpp [hunk 5]
```diff
@@ -85,27 +82,35 @@ void task4()
...
-		std::cout << std::string(i , '*') << '\n';
-
-	std::cout << "\n";
-
-	for (std::size_t i = 0; i < length; ++i)
-		std::cout << std::string(i, ' ') + std::string(length, '*') << '\n';
+	//a
+	{
+		for (std::size_t i{ 0 }; i < length; ++i)
+			std::cout << std::string(i + 1, '*') << '\n';
```
Comment (helen-black) [inline Hw5/main.cpp:94]: It's better to use brackets even with one line loop or if

**Review by helen-black — CHANGES_REQUESTED** [review]

File: Hw5/main.cpp (hunk 5 above)
Comment (VolodiaH) [reply Hw5/main.cpp:94]: I changed it, but brackets with one operation seem excessive and redundant, especially with trivial functions which are not error-prone

File: Hw5/main.cpp (hunk 4 above)
Comment (VolodiaH) [reply Hw5/main.cpp]: Changed.

**Review by helen-black — APPROVED** [review]

**Pull request merged**